mne/viz/evoked.py

```diff
@@ -1440,21 +1442,153 @@ def _truncate_yaxis(axes, ymin, ymax, orig_ymin, orig_ymax, fraction,
-                         linestyles=['-'], styles=None, vlines=list((0.,)),
-                         ci=0.95, truncate_yaxis=False, truncate_xaxis=True,
-                         ylim=dict(), invert_y=False, show_sensors=None,
-                         show_legend=True, axes=None, title=None, show=True):
-    """Plot evoked time courses for one or multiple channels and conditions.
+def _combine_grad(evoked, picks):
+    """Create a new instance of Evoked with combined gradiometers (RMSE)."""
+    def pair_and_combine(data):
+        data = data ** 2
+        data = (data[::2, :] + data[1::2, :]) / 2
+        return np.sqrt(data)
+    picks, ch_names = _grad_pair_pick_and_name(evoked.info, picks)
+    this_data = pair_and_combine(evoked.data[picks, :])
+    ch_names = ch_names[::2]
+    evoked = evoked.copy().pick_channels(ch_names)
+    combined_ch_names = [ch_name[:-1] + "X" for ch_name in ch_names]
+    evoked.rename_channels({c_old: c_new for c_old, c_new
+                            in zip(evoked.ch_names, combined_ch_names)})
+    evoked.data = this_data
+    return evoked
+
+
+def _check_loc_legal(loc, what='your choice'):
+    """Check if loc is a legal location for MPL subordinate axes."""
+    true_default = {"show_legend": 3, "show_sensors": 4}.get(what, 1)
+    loc_dict = {'upper right': 1, 'upper left': 2, 'lower left': 3,
+                'lower right': 4, 'right': 5, 'center left': 6,
+                'center right': 7, 'lower center': 8, 'upper center': 9,
+                'center': 10, True: true_default}
+    loc_ = loc_dict.get(loc, loc)
+    if loc_ not in range(11):
+        raise ValueError(str(loc) + " is not a legal MPL loc, please supply"
+                         "another value for " + what + ".")
+    return loc_
+
+
+def _format_evokeds_colors(evokeds, cmap, colors):
+    """Set up to have evokeds as a dict as well as colors."""
+    from ..evoked import Evoked, _check_evokeds_ch_names_times
 
-    This function is useful for comparing ER[P/F]s at a specific location. It
-    plots Evoked data or, if supplied with a list/dict of lists of evoked
-    instances, grand averages plus confidence intervals.
+    if isinstance(evokeds, Evoked):
+        evokeds = dict(Evoked=evokeds)  # title becomes 'Evoked'
+    elif not isinstance(evokeds, dict):  # it's assumed to be a list
+        if (cmap is not None) and (colors is None):
+            colors = dict((str(ii + 1), ii) for ii, _ in enumerate(evokeds))
+        evokeds = dict((str(ii + 1), evoked)
+                       for ii, evoked in enumerate(evokeds))
+    else:
+        assert isinstance(evokeds, dict)
+        if (colors is None) and cmap is not None:
+            raise ValueError('If evokeds is a dict and a cmap is passed, '
+                             'you must specify the colors.')
+            # XXX : I am a bit concerned about the duplication of
+            # the colors and cmap parameters.
```

Comment:
@jona-sassenhagen what do you think about this?
we don't use colors anywhere in doc and examples and it has a non-trivial interaction with cmap.

thoughts?

Reply:
Hm? Maybe I misunderstand, but `colors` is all over the example.

`colors` specifies the mapping from conditions to line colors relatively to the color map, `cmap` specifies the colormap (and the name for the colorbar). I see two alternatives, both of which we ruled out:

- use magic to read relative line color from the condition names (my original idea, which you ruled out)
- demand `colors` values to be RGB or RGBA (possible, but requires more user work)

Essentially, option two is viable, but would require the user do something like

```
import seaborn as sns
from sklearn.preprocessing import QuantileTransformer

n_bins = 10
qt = QuantileTransformer(n_bins).fit(epochs.metadata[cond])
colorscale = sns.color_palette("viridis", n_bins)

colors = dict()
for ... in ...:
    colors[xxx] = colorscale[qt.transform(...)]
```

every time they run the function.

Reply:
ok ok. forget it.

Reply:
should the comment be removed, then?